src/main/java/org/apache/sling/cpconverter/maven/mojos/ContentPackage.java

```diff
@@ -90,13 +99,57 @@ public boolean isModuleIsContentPackage() {
             // all dependencies, transitives included
             artifacts = project.getArtifacts();
         }
+
+        // Sometimes artifacts don't have their 'file' attribute set, which we need. For those that don't, resolve them
+        final Set<Artifact> fileArtifacts = new HashSet<>();
+        for (Artifact a : artifacts) {
+            if (a.getFile() != null) {
+                fileArtifacts.add(a);
+            } else {
+                if (repoSystem != null && repoSession != null) {
+                    // Resolving the artifact via Aether will fill in the file attribute
+                    Artifact fileArt = resolveArtifact(repoSystem, repoSession, a);
+                    if (fileArt != null) {
```

Comment:
Is it an error condition if the file can't be resolved?

Reply:
I think it would be an error condition in reality. Right now the code ignores those artifacts, but maybe it would be better to throw an exception instead...

Reply:
Yes, that was my thinking here as well, better throw an exception

Apart from that the PR looks good

Reply:
I've made that change. Thanks for the review, @cziegeler !